src/_guides/libraries/objective-c-interop.md

```diff
@@ -293,6 +293,58 @@ then check the status, and wait for the duration of the audio file:
     }
 ```
 
+### Callbacks and multithreading limitations
+
+Multithreading issues are the biggest limitation
+of Dart's experimental support for Objective-C interop.
+While `ffigen` does support converting
+Dart functions to Objective-C blocks,
+these limitations must be kept in mind when using them.
+These issues also affect the use of certain Apple APIs.
+
+1. Dart isolates are not the same thing as threads.
+   An isolate is not guaranteed to run on a particular thread,
+   and the VM may change which thread an isolate is running on
+   without warning.
+   There is an [open feature request][] to allow isolates to be
+   pinned to specific theads.
+2. Most Apple APIs don't have any guarantees about
+   which thread a callback will be run on.
+3. Most APIs that involve UI interaction
+   can only be called on the main thread,
+   also known in Flutter as the platform thread.
+4. Many Apple APIs are [not thread safe][].
+
+Points 1 and 2 mean that a callback created in one isolate
+may be invoked on a thread running a different isolate,
+or no isolate at all.
+This will cause your app to crash.
```

Comment:
```suggestion
The first two points mean that a callback created in one isolate
might be invoked on a thread running in a different isolate,
or in no isolate at all.
This will cause your app to crash.
```

The phrase "...a callback created in one isolate might be invoked on a thread running in a different isolate..."  confuses me. Do you mean to say that it might be invoked on a different isolate?

Reply:
Isolates run on threads, not the other way round. So the phrasing suggested here, "a thread running in a different isolate", is inaccurate. But maybe if I explain the situation more fully to you, you can help boil it down to something our users will follow.

Basically, the app has multiple isolates, and multiple threads that run those isolates. Each thread can only run one isolate at a time, or can be running other non-Dart code that's not part of an isolate. Additionally, an isolate can be run on one thread, then paused, and later resumed on a different thread.

When a user converts a Dart function to an Objective-C block, the Dart code inside that block is permanently associated with the isolate it was created in. So if Objective-C later runs that block on a different thread that is running the wrong isolate, or no isolate, or runs it on the same thread but the isolate has moved to a different thread, it will crash.

Reply:
Reworded point 1 to make this a bit clearer. I could also make a diagram if you think this is still unclear.